server/controllers/finance/purchases.js

```diff
@@ -533,6 +535,31 @@ function purchaseBalance(req, res, next) {
     .done();
 }
 
+/**
+ * @function remove
+ *
+ * @decription
+ * This function allows users to delete purchase orders that are awaiting confirmation.
+ */
+async function remove(req, res, next) {
+  const pid = db.bid(req.params.uuid);
+
+  try {
+    const record = await db.one('SELECT * FROM purchase WHERE uuid = ?', pid);
+
+    const WAITING_CONFIRMATION = 1;
+    if (record.status_id !== WAITING_CONFIRMATION) {
+      throw new BadRequest('Cannot remove a purchase order that is confirmed');
```

Comment:
Since I presume the status_id could contain more than 2 values (otherwise it would be a boolean, right?), it seems like this bad request warning should be something like:  "Can only remove purchases that have not been confirmed".

Reply:
Yeah, this a good suggestion.  I'll make the change.